clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -44,12 +60,58 @@ interface FossIdRestService {
         /**
          * The mapper for JSON (de-)serialization used by this service.
          */
-        val JSON_MAPPER = JsonMapper()
+        val JSON_MAPPER: ObjectMapper = JsonMapper()
             .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
             // FossID has a bug in get_results/scan.
             // Sometimes the match_type is "ignored", sometimes it is "Ignored".
-            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
-            .registerKotlinModule()
+            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS).copy()
+            .registerModule(
+                kotlinModule()
+                    .setDeserializerModifier(FossIdDeserializerModifier())
+            )
+
+        /**
+         * The FossID server is not consistent. It usually returns a List or a Map, but if there is no
+         * entry, it returns data:false.
+         * We register two custom deserializers to address this issue.
```

Comment:
I think the comment of this class could be improved. I would use something like this:

"A class to modify the standard Jackson deserialization to deal with inconsistencies in responses sent by the FossID server.

FossID usually returns a List or a Map, but if there is no entry, it returns data:false. To handle this special case, this class makes sure that a custom deserializer is used for lists or maps that detects such a result and converts it to an empty list or map, respective. That way, the FossID service interface can use meaningful result types."